src/mscorlib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -324,6 +324,11 @@ internal unsafe static CustomAttributeRecord[] GetCustomAttributeRecords(Runtime
             MetadataEnumResult tkCustomAttributeTokens;
             scope.EnumCustomAttributes(targetToken, out tkCustomAttributeTokens);
 
+            if (tkCustomAttributeTokens.Length == 0)
+            {
+                return Array.Empty<CustomAttributeRecord>();
```

Comment:
[`Array.Empty<T>`](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Array.cs#L1054-L1061) is a static [empty array of type T](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Array.cs#L2834-L2837) shared among the entire framework, rather than each type having their own `static new T[0]`